src/test/tests.ts

```diff
@@ -809,15 +805,16 @@ describe('Server', function () {
 
     setTimeout(() => {
       expect(connectionCallbackSpy.calledOnce).to.be.true;
-      expect(connectionCallbackSpy.getCall(0).args[0]).to.equal('Error');
+      expect(connectionCallbackSpy.getCall(0).args[0].message).to.equal('Error');
```

Comment:
What kind of payload errors we can expect to come inside our mapping function, and how they should came out in the end in legacy mode? Also how do u think we should expose such legacy flag? Should we make it a public class property? At the moment it is bind to connectionContext and we can access connectionContext into onConnect callback, we can also change it this way. What do u think it's the best way @DxCx ?